tests/unit/ssg-module/test_id_translate.py

```diff
@@ -21,3 +29,214 @@ def test__split_namespace():
+    expected_object_id = "oval:ssg-obj_kerberos_disable_no_keytab:obj:1"
+    expected_state_id = "oval:ssg-filter_ssh_key_owner_root:ste:1"
+    new_tree = idtranslator.translate(oval_tree)
+    real = _parse_interesting_ids(new_tree)
+    assert real.definition_id == expected_definition_id
+    assert real.criterion_test_ref == expected_test_id
+    assert real.criterion_test_ref == real.test_id
+    assert real.test_id == expected_test_id
+    assert real.object_ref == expected_object_id
+    assert real.object_ref == real.object_id
+    assert real.object_id == expected_object_id
+    assert real.filter_ref == expected_state_id
+    assert real.filter_ref == real.state_id
+    assert real.state_id == expected_state_id
+
+
+def test_idtranslator_translate_oval_xmldiff(idtranslator, oval_tree):
+    xmldiff_main = pytest.importorskip("xmldiff.main")
+    xmldiff_actions = pytest.importorskip("xmldiff.actions")
+    LET = pytest.importorskip("lxml.etree")
+    old = LET.fromstring(ET.tostring(oval_tree))
+    new = LET.fromstring(ET.tostring(idtranslator.translate(oval_tree)))
+    inverted_new_nsmap = {v: k for k, v in old.nsmap.items()}
+    o = inverted_new_nsmap[oval_namespace]
+    ou = inverted_new_nsmap[oval_unix_namespace]
+    real_diff = set(xmldiff_main.diff_trees(old, new))
+    expected_diff = {
+        xmldiff_actions.UpdateAttrib(
+            node=f'/{o}:oval_definitions/{o}:definitions/{o}:definition[1]',
+            name='id',
+            value='oval:ssg-kerberos_disable_no_keytab:def:1'),
+        xmldiff_actions.UpdateAttrib(
+            node=f'/{o}:oval_definitions/{o}:tests/{ou}:file_test[1]',
+            name='id',
+            value='oval:ssg-test_kerberos_disable_no_keytab:tst:1'),
+        xmldiff_actions.UpdateAttrib(
+            node=f'/{o}:oval_definitions/{o}:objects/{ou}:file_object[1]',
+            name='id',
+            value='oval:ssg-obj_kerberos_disable_no_keytab:obj:1'),
+        xmldiff_actions.UpdateAttrib(
+            node=f'/{o}:oval_definitions/{o}:states/{ou}:file_state[1]',
+            name='id',
+            value='oval:ssg-filter_ssh_key_owner_root:ste:1'),
+        xmldiff_actions.UpdateAttrib(
+            node=(
+                f'/{o}:oval_definitions/{o}:tests/{ou}:file_test/'
+                f'{ou}:object[1]'),
+            name='object_ref',
+            value='oval:ssg-obj_kerberos_disable_no_keytab:obj:1'),
+        xmldiff_actions.UpdateTextIn(
+            node=(
+                f'/{o}:oval_definitions/{o}:objects/{ou}:file_object/'
+                f'{o}:filter[1]'),
+            text='oval:ssg-filter_ssh_key_owner_root:ste:1'),
+        xmldiff_actions.UpdateAttrib(
+            node=(
+                f'/{o}:oval_definitions/{o}:definitions/{o}:definition/'
+                f'{o}:criteria/{o}:criterion[1]'),
+            name='test_ref',
+            value='oval:ssg-test_kerberos_disable_no_keytab:tst:1')
+    }
```

Comment:
@matejak said that it would be better if the expected items are removed from the diff and finally it would be checked if the diff is empty.